nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/util/PasswordUtil.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.toolkit.tls.util;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+
+public class PasswordUtil {
+    private final SecureRandom secureRandom;
+
+    public PasswordUtil(SecureRandom secureRandom) {
+        this.secureRandom = secureRandom;
+    }
+
+    public String generatePassword() {
+        // [see http://stackoverflow.com/questions/41107/how-to-generate-a-random-alpha-numeric-string#answer-41156]
+        return new BigInteger(1024, secureRandom).toString(36);
```

Comment:
I appreciate the *strong* entropy here, but 1024 bits in base 36 &asymp; 198 characters. Something that provides more than 128 bits is sufficient here; 256 will be more than enough for a number of years. I might have commented off-hand about the base earlier, but base 32 makes sense because it uses a "round" number of bits  ( `2^5 = 32` ) to fully express its alphabet, while base 36 (`0-9a-f`) needs 6 ( `2^6 = 64` ) but wastes a number of those bits.